src/main/java/eu/dissco/backend/repository/ElasticSearchRepository.java

```diff
@@ -24,33 +26,46 @@
 @RequiredArgsConstructor
 public class ElasticSearchRepository {
 
-
-  private static final String INDEX = "new-dissco";
-  private static final String ANNOTATION_EXISTS_QUERY = "_exists_:annotation.type ";
+  private static final String DIGITAL_SPECIMEN_INDEX = "digital-specimen";
+  private static final String ANNOTATION_INDEX = "annotation";
   private static final String FIELD_CREATED = "created";
   private static final String FIELD_GENERATED = "generated";
   private final ElasticsearchClient client;
-  private final DateTimeFormatter formatter;
-
-  private final Pattern pattern = Pattern.compile("-?\\d+(\\.\\d+)?");
 
-  public List<DigitalSpecimen> search(String query, int pageNumber, int pageSize)
+  public List<DigitalSpecimen> search(Map<String, List<String>> params, int pageNumber,
+      int pageSize)
       throws IOException {
-    query = query.replace("/", "//");
-
     var offset = getOffset(pageNumber, pageSize);
-
-    var searchRequest = new SearchRequest.Builder().index(INDEX)
-        .q("_exists_:digitalSpecimen.physicalSpecimenId AND " + query).from(offset).size(pageSize)
-        .build();
+    var queries = generateQueries(params);
+    var searchRequest = new SearchRequest.Builder().index(DIGITAL_SPECIMEN_INDEX)
+        .query(
+            q -> q.bool(b -> b.should(queries).minimumShouldMatch(String.valueOf(params.size()))))
+        .from(offset)
+        .size(pageSize).build();
     return client.search(searchRequest, ObjectNode.class).hits().hits().stream().map(Hit::source)
         .map(this::mapToDigitalSpecimen).toList();
   }
 
+  private List<Query> generateQueries(Map<String, List<String>> params) {
+    var queries = new ArrayList<Query>();
+    for (var entry : params.entrySet()) {
+      for (var value : entry.getValue()) {
+        Query query;
+        if (Objects.equals(entry.getKey(), "q")) {
+          var sanitisedValue = value.replace("/", "//");
+          query = new Query.Builder().queryString(q -> q.query(sanitisedValue)).build();
+        } else {
+          query = new Query.Builder().term(t -> t.field(entry.getKey()).value(value)).build();
+        }
+        queries.add(query);
+      }
+    }
+    return queries;
+  }
+
```

Comment:
So, this is where you build the query for ElasticSearch from the given params right?

Reply:
Yes, so we use the boolean queries from elastic.
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html
We use the should query.
Each term is a should and for which one value should match.
So we got a bunch of shoulds and the `minimumShouldMatch` indicates how many shoulds shoud match

Reply:
Yes, so thanks to the should clause it can fetch multiple records right? With must it would to be focused on a certain specimen and the filter clause is something in between I guess, it says is not dependent on the score of the search? (Don't really get that one).

Reply:
Yes so we always get back an array. Even for physicalSpecimenId which should always return 1 record. The score, what I read this will be used. So the more the should hit the higher the score. This is mainly important with for the free text as it could partly match. Most are exact matches which is either true or false.
For documentation:
`The bool query takes a more-matches-is-better approach, so the score from each matching must or should clause will be added together to provide the final _score for each document.`